src/components/AssetSelectModal/AssetSelectModal.tsx

```diff
@@ -0,0 +1,159 @@
+import {
+  Button,
+  CloseButton,
+  Flex,
+  Input,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+} from '@chakra-ui/react'
+import type { ChainId } from '@shapeshiftoss/caip'
+import type { ChangeEvent } from 'react'
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react'
+import AssetData from 'lib/generatedAssetData.json'
+import { isNft } from 'lib/utils'
+import type { Asset } from 'types/Asset'
+
+import { AssetList } from './AssetList'
+import type { ChainRow } from './ChainButton'
+import { ChainButton } from './ChainButton'
+import { filterAssetsBySearchTerm } from './helpers/filterAssetsBySearchTerm'
+
+type AssetSelectModalProps = {
+  isOpen: boolean
+  onClose: () => void
+  onClick: (asset: Asset) => void
+}
+
+export const AssetSelectModal: React.FC<AssetSelectModalProps> = ({ isOpen, onClose, onClick }) => {
+  const [searchQuery, setSearchQuery] = useState('')
+  const assets = Object.values(AssetData)
```

Comment:
Even though `generatedAssetData.json` is most likely temporary and we may use data from THOR instead (or a sane subset of the `generatedAssetData.json`), this is a hyuge guy to parse so `useMemo` pls 🙏🏽